format.go

```diff
@@ -4,126 +4,144 @@ import (
-func Unpack(err error) UnpackedError {
-	var upErr UnpackedError
-	for err != nil {
-		switch err := err.(type) {
-		case *rootError:
-			upErr.ErrRoot.Msg = err.msg
-			upErr.ErrRoot.Stack = err.stack.get()
-		case *wrapError:
-			// prepend links in stack trace order
-			link := ErrLink{Msg: err.msg}
-			link.Frame = err.frame.get()
-			upErr.ErrChain = append([]ErrLink{link}, upErr.ErrChain...)
-		default:
-			upErr.ExternalErr = err.Error()
-		}
-		err = Unwrap(err)
-	}
-	return upErr
+// FormatOptions defines output options like omitting stack traces and inverting the error or stack order.
+type FormatOptions struct {
+	InvertOutput bool // Flag that inverts the error output (wrap errors shown first).
+	WithTrace    bool // Flag that enables stack trace output.
+	InvertTrace  bool // Flag that inverts the stack trace output (top of call stack shown first).
+	// todo: maybe allow users to hide wrap frames if desired
 }
 
-// Format defines an error output format to be used with the default formatter.
-type Format struct {
-	WithTrace    bool   // Flag that enables stack trace output.
-	MsgStackSep  string // Separator between error messages and stack frame data.
-	PreStackSep  string // Separator at the beginning of each stack frame.
-	StackElemSep string // Separator between elements of each stack frame.
-	ErrorSep     string // Separator between each error in the chain.
+// StringFormat defines a string error format.
+type StringFormat struct {
+	Options      FormatOptions // Format options (e.g. omitting stack trace or inverting the output order).
+	MsgStackSep  string        // Separator between error messages and stack frame data.
+	PreStackSep  string        // Separator at the beginning of each stack frame.
+	StackElemSep string        // Separator between elements of each stack frame.
+	ErrorSep     string        // Separator between each error in the chain.
 }
 
-// NewDefaultFormat conveniently returns a basic format for the default string formatter.
-func NewDefaultFormat(withTrace bool) Format {
-	stringFmt := Format{
-		WithTrace: withTrace,
-		ErrorSep:  ":",
+// NewDefaultStringFormat returns a default string output format.
+func NewDefaultStringFormat(options FormatOptions) StringFormat {
+	stringFmt := StringFormat{
+		Options: options,
 	}
-	if withTrace {
+	if options.WithTrace {
 		stringFmt.MsgStackSep = "\n"
 		stringFmt.PreStackSep = "\t"
 		stringFmt.StackElemSep = ":"
 		stringFmt.ErrorSep = "\n"
+	} else {
+		stringFmt.ErrorSep = ": "
 	}
 	return stringFmt
 }
 
-// ToCustomString returns a custom formatted string for a given eris error.
+// ToString returns a default formatted string for a given error.
 //
-// To declare custom format, the Format object has to be passed as an argument.
 // An error without trace will be formatted as following:
+//   <Root error msg>: <Wrap error msg>
+// An error with trace will be formatted as following:
+//   <Root error msg>
```

Comment:
do we really need these comments showing separators? it's covered pretty well in our readme already so i'd like to eliminate this maintenance if possible

Reply:
I will suggest to keep it until next release and then delete it. It helps when editor shows autocomplete

Reply:
sounds good, in that case, i need to update the order of them. will do that in a bit.

Reply:
should be fixed now